File: app/src/main/java/wangdaye/com/geometricweather/background/receiver/MainReceiver.java
package wangdaye.com.geometricweather.background.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.text.TextUtils;

import wangdaye.com.geometricweather.utils.helpter.IntentHelper;

/**
 * Main receiver.
 * */

public class MainReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        String action = intent.getAction();
        if (TextUtils.isEmpty(action)) {
            return;
        }
        switch (action) {
            case Intent.ACTION_BOOT_COMPLETED:
            case Intent.ACTION_WALLPAPER_CHANGED:
                IntentHelper.startAwakeForegroundUpdateService(context);
                break;
        }
    }
}

File: app/src/main/java/org/breezyweather/background/receiver/MainReceiver.kt
package org.breezyweather.background.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import org.breezyweather.background.weather.WeatherUpdateJob

/**
 * Main receiver.
 */
class MainReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val action = intent.action
        if (action.isNullOrEmpty()) return
        when (action) {
            // TODO: Do we really need this?
            Intent.ACTION_BOOT_COMPLETED, Intent.ACTION_WALLPAPER_CHANGED -> WeatherUpdateJob.startNow(context)
        }
    }
}